src/bulk-load.js

```diff
@@ -115,6 +116,64 @@ module.exports = class BulkLoad extends EventEmitter {
     }
   }
 
+  /**
+    @param {Object} bulkLoadOptions
+    @param {boolean} bulkLoadOptions.checkConstraints - honors constraints during bulk load, it is disabled by default.
+    @param {boolean} bulkLoadOptions.fireTriggers - honors insert triggers during bulk load, it is disabled by default.
+    @param {boolean} bulkLoadOptions.keepNulls - honors null value passed, ignores the default values set on table.
+    @param {boolean} bulkLoadOptions.tableLock- places a bulk update(BU) lock on table while performing bulk load. Uses row locks by default.
+  */
+  setOptions(bulkLoadOptions) {
+    if (typeof bulkLoadOptions !== 'object') {
+      throw new TypeError('"bulkLoadOptions" argument must be an object');
+    }
+
+    const bOptions = {};
+    bOptions.checkConstraints = getBooleanOption('checkConstraints', false);
```

Comment:
I believe the current version of `setOptions` behaves differently than your initial version, and differently than how I as a user would expect it to behave.

I'd expect multiple `setOptions` calls to only change the options referred by the keys of the `options` object that is passed in, like this:

```
bulkLoad.setOptions({ fireTrigger: true });
bulkLoad.setOptions({ keepNulls: true });
// => bulkLoad should have both fireTriggers and keepNulls enabled
```

But I don't think this is how it works now. 😞 

@v-suhame What do you think would be the correct approach here?

Reply:
@arthurschreiber I would expect that `setOptions()` replaces the previous options and that the missing options get the default value. Otherwise I would expect it to be named `addOptions()` or `updateOptions()`.

If the options could be passed with `connection.newBulkLoad()`, we would not need `setOptions()` at all.

Reply:
>// => bulkLoad should have both fireTriggers and keepNulls enabled

Making this change would affect existing users. Its best to let the user decide if they want to enable these options, and moreover it is the same behavior as other SQL Server drivers.  

>I would expect that setOptions() replaces the previous options and that the missing options get the default value.

@arthurschreiber This was the intended behavior after we moved from `addOptions()` 👆.

Reply:
@v-suhame What do you think about the suggestion from @chdh to move the options to the newbulkload method instead of having a setOptions method?

Reply:
Are we looking for something like the below snippet after change? 

````node
  connection.newBulkLoad(tableName, {
      callback: cb,
      bulkLoadOptions : {
        keepNulls: true,
        checkConstraints: true
        ..
      }
  });
```

Reply:
@v-suhame Yes, or simpler with all options mixed at the same level:

```
connection.newBulkLoad(tableName, {
      callback: cb,
      keepNulls: true,
      checkConstraints: true
      ..
  });
```

Reply:
@v-suhame @chdh I'd prefer if we could stick to the convention of having the callback as the last argument of the function, instead of being part of an options object. 🙇

We'd also need to keep backwards compatibility. What about something like this:

```js
newBulkLoad(table, options, callback) {
  if (callback === undefined) {
     callback = options;
     options = undefined;
  }

  return new BulkLoad(table, this.config.options, options, callback);
}
```

and then in `bulk-load.js`:

```js
  constructor(table, connectionOptions, {
    checkConstraints = false,
    fireTriggers = false,
    keepNulls = false,
    lockTable = false,
  }, callback) {
    super();

    if (typeof checkConstraints !== 
'boolean') {
      throw new TypeError('The "options.checkConstraints" property must be of type boolean.');
}

    if (typeof fireTriggers !== 
'boolean') {
      throw new TypeError('The "options.fireTriggers" property must be of type boolean.');
}

    if (typeof keepNulls !== 
'boolean') {
      throw new TypeError('The "options.keepNulls" property must be of type boolean.');
}

    if (typeof lockTable !== 
'boolean') {
      throw new TypeError('The "options.lockTable" property must be of type boolean.');
}

    this.error = undefined;
    this.canceled = false;

    this.table = table;
    this.options = connectionOptions;
    this.callback = callback;
    this.columns = [];
    this.columnsByName = {};
    this.rowsData = new WritableTrackingBuffer(1024, 'ucs2', true);
    this.firstRowWritten = false;

    this.bulkOptions = { checkConstraints, fireTriggers, keepNulls, lockTable };
  }
```